### ISSUE TITLE
Failed to install 'cordova-plugin-app-preferences': Error: ENOENT: no such file or directory

### ISSUE THREAD

**Issue by nghtstr** [apla/me.apla.cordova.app-preferences#135]
Issue: Every time I try to install this plugin, I keep getting these errors. Here is one snippet of what I keep on getting:

`Plugin "cordova-plugin-app-preferences" already installed on android.
Installing "cordova-plugin-app-preferences" for browser
unhandled exception { Error: ENOENT: no such file or directory, open 'platforms/android/src/me/apla/cordova/AppPreferencesActivity.java'
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: 'platforms/android/src/me/apla/cordova/AppPreferencesActivity.java' }
Failed to install 'cordova-plugin-app-preferences': Error: ENOENT: no such file or directory, open 'platforms/android/src/me/apla/cordova/AppPreferencesActivity.java'
(node:8160) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: ENOENT: no such file or directory, open 'platforms/android/src/me/apla/cordova/AppPreferencesActivity.java'
(node:8160) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
`

After deleting it manually, I try doing a clean install of it:

`Installing "cordova-plugin-app-preferences" for android
Android Studio project detected
unhandled exception { Error: ENOENT: no such file or directory, open 'platforms/android/src/me/apla/cordova/AppPreferencesActivity.java'
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: 'platforms/android/src/me/apla/cordova/AppPreferencesActivity.java' }
Failed to install 'cordova-plugin-app-preferences': Error: ENOENT: no such file or directory, open 'platforms/android/src/me/apla/cordova/AppPreferencesActivity.java'
(node:8196) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: ENOENT: no such file or directory, open 'platforms/android/src/me/apla/cordova/AppPreferencesActivity.java'
(node:8196) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
`

As you can see, there is no difference of when I try to install this, or reinstall this.
_____

Please specify your environment

Plugin version:
  - [X] released version: 0.99.3
  - [ ] repository master

Toolchain:
  - [X] Cordova cli
  - [ ] Phonegap cli
  - [ ] Phonegap cloud
  - [ ] Ionic
  - [ ] Other: _____

Platforms affected:
  - [X] Android
  - [X] iOS/macOS
  - [X] LocalStorage fallback for browser and blackberry
  - [ ] Windows and Windows Phone 8.1 and later
  - [ ] Windows Phone 8 and earlier (deprecated)

What the scope of your problem:
  - [ ] General functionality (store/fetch/remove/clearAll)
  - [ ] Suites
  - [ ] Cloud synchronization and events
  - [ ] Preferences pane generation and display
  - [X] Installation

**Comment by dkijkuit** [comment]
Having the same issue here, any ideas?

**Comment by iamjoyce** [comment]
What is your cordova version? My initial suspicion is that the cordova version is `7.0.0` or `8.0.0` - The cordova version affected few other plugins as well. Downgrading should help.

**Comment by dkijkuit** [comment]
@iamjoyce yes version is indeed `7.0.0`, I wasn't aware that these versions weren't supported yet.

**Comment by Story5** [comment]
I having the same issue,blow is my ionic info
```
cli packages: (/usr/local/lib/node_modules)

    @ionic/cli-utils  : 1.19.1
    ionic (Ionic CLI) : 3.19.1

global packages:

    cordova (Cordova CLI) : 8.0.0 

local packages:

    @ionic/app-scripts : 3.1.2
    Cordova Platforms  : android 7.0.0 ios 4.5.4
    Ionic Framework    : ionic-angular 3.9.2

System:

    Android SDK Tools : 26.1.1
    ios-deploy        : 1.9.2 
    ios-sim           : 6.1.2 
    Node              : v8.9.4
    npm               : 5.6.0 
    OS                : macOS High Sierra
    Xcode             : Xcode 9.2 Build version 9C40b 

Environment Variables:

    ANDROID_HOME : /Users/Story5/Library/Android/sdk

Misc:

    backend : pro
```

**Comment by fabiocaccamo** [comment]
Any news about this issue?

**Comment by nkCreation** [comment]
Hi. Any news of this ?

**Comment by mapicard** [comment]
Having the same issue here. How can we get an updated version of the plugin?

**Comment by jacobg** [comment]
The forked version at https://github.com/vash15/me.apla.cordova.app-preferences fixes the issue. Please merge it in.

**Comment by mapicard** [comment]
Thanks for the reply, but pretty much of a newbie here.
I know how to manage/merge my git but how do I "merge it in" ? 
Do I just download and copy all the files from this fork ? 
Do I install the plugin from this git-repo ? 
Something else ?
Thanks again
mp

**Comment by jacobg** [comment]
:) I'm asking the owner of this repository to merge it in.

**Comment by unstableair** [comment]
@mapicard to install the patched version use:

`cordova plugin add https://github.com/vash15/me.apla.cordova.app-preferences`

**Comment by fdambrosio** [comment]
thanks @unstableair  this works

**Comment by virtualvishwam** [comment]
@owner  please patch this... Its been too long and I am just waiting for this plugin to work ( not able to migrate to Cordova 8.0.0).
@jacobg lemme try the patch... Thanks in advance...

**Comment by jamescodesthings** [comment]
@jacobg Because of the reverse domain naming convention in the repository name plugman kept giving me 'Could not fetch <plugin> from registry' errors. This caused trouble when installing from scratch on our CI server.

To fix this I've re-hosted your fork here: https://bitbucket.org/packt-internal/cordova-plugin-app-preferences/src/master/

With the same repo-name as the plugin name, that tends to work, if anyone has the same trouble when reinstalling platforms or running `cordova prepare` then use this hosted copy with:

`cordova plugin add https://bitbucket.org/packt-internal/cordova-plugin-app-preferences.git`

Thanks for sorting out Cordova 7+ support.

**Comment by fdambrosio** [comment]
thanks @jamescodesthings

**Comment by gwynjudd** [comment]
@jacobg please submit a PR for your fixes

**Comment by borgogelli** [comment]
Same problem for me

**Comment by rejhgadellaabsl** [comment]
+1 here. Is this project still maintained? I assume @jacobg did PR his fix?

**Comment by mamzellejuu** [comment]
+1 thanks @jacobg

**Comment by theCrius** [comment]
Issue is still there:

Ionic 3.9, cordova 7.1

had to use @jamescodesthings source to have it working

**Comment by mayursofbang4** [comment]
I am able to open the app preferences in IOS via _**Cordova plugin add https://github.com/myieye/me.apla.cordova.app-preferences**_ but not working in android. Can anyone have a solution for this issue? 

We are getting an error while clicking on the app preference icon - _**Class me.apla.cordova.AppPreferencesActivity not found. Please run preference generator.**_


[cordova plugin add https://github.com/myieye/me.apla.cordova.app-preferences](url)

**Comment by josepoma** [comment]
I have the same problem that you report @mayursofbang4
